types/testingutils/beacon_node_versioned.go

```diff
@@ -0,0 +1,167 @@
+	ForkSlotCapella = 6209536 // Epoch(194048)
+	// TestingDutySlotBellatrix keeping this value to not break the test roots
+	TestingDutySlotBellatrix = 12
+)
+
+var TestingBeaconBlockV = func(version spec.DataVersion) *spec.VersionedBeaconBlock {
+	switch version {
+	case spec.DataVersionBellatrix:
+		return &spec.VersionedBeaconBlock{
+			Version:   version,
+			Bellatrix: TestingBeaconBlock,
+		}
+	default:
+		panic("unsupported version")
+	}
+}
+
+var TestingBeaconBlockBytesV = func(version spec.DataVersion) []byte {
+	var ret []byte
+	vBlk := TestingBeaconBlockV(version)
+	if vBlk.IsEmpty() {
+		panic("empty block")
+	}
+
+	switch version {
+	case spec.DataVersionBellatrix:
+		ret, _ = vBlk.Bellatrix.MarshalSSZ()
+
+	default:
+		panic("unsupported version")
+	}
+
+	return ret
+}
+
+var TestingBlindedBeaconBlockV = func(version spec.DataVersion) *api.VersionedBlindedBeaconBlock {
+	switch version {
+	case spec.DataVersionBellatrix:
+		return &api.VersionedBlindedBeaconBlock{
+			Version:   version,
+			Bellatrix: TestingBlindedBeaconBlock,
+		}
+	default:
+		panic("unsupported version")
+	}
+}
+
+var TestingBlindedBeaconBlockBytesV = func(version spec.DataVersion) []byte {
+	var ret []byte
+	vBlk := TestingBlindedBeaconBlockV(version)
+	if vBlk.IsEmpty() {
+		panic("empty block")
+	}
+
+	switch version {
+	case spec.DataVersionBellatrix:
+		ret, _ = vBlk.Bellatrix.MarshalSSZ()
+
+	default:
+		panic("unsupported version")
+	}
+
+	return ret
+}
+
+var TestingWrongBeaconBlockV = func(version spec.DataVersion) *spec.VersionedBeaconBlock {
+	blkByts := TestingBeaconBlockBytesV(version)
+
+	switch version {
+	case spec.DataVersionBellatrix:
+		ret := &bellatrix.BeaconBlock{}
+		if err := ret.UnmarshalSSZ(blkByts); err != nil {
+			panic(err.Error())
+		}
+		ret.Slot = 100
+		return &spec.VersionedBeaconBlock{
+			Version:   version,
+			Bellatrix: ret,
+		}
+
+	default:
+		panic("unsupported version")
+	}
+}
+
+var TestingSignedBeaconBlockV = func(ks *TestKeySet, version spec.DataVersion) ssz.HashRoot {
+	vBlk := TestingBeaconBlockV(version)
+	if vBlk.IsEmpty() {
+		panic("empty block")
+	}
+
+	switch version {
+	case spec.DataVersionBellatrix:
+		return &bellatrix.SignedBeaconBlock{
+			Message:   vBlk.Bellatrix,
+			Signature: signBeaconObject(vBlk.Bellatrix, types.DomainProposer, ks),
+		}
+
+	default:
+		panic("unsupported version")
+	}
+}
+
+var VersionBySlot = func(slot phase0.Slot) spec.DataVersion {
+	if slot < ForkSlotCapella {
+		return spec.DataVersionBellatrix
+	}
+	return spec.DataVersionCapella
+}
+
+var TestingProposerDutyV = func(version spec.DataVersion) *types.Duty {
+	duty := &types.Duty{
+		Type:                    types.BNRoleProposer,
+		PubKey:                  TestingValidatorPubKey,
+		ValidatorIndex:          TestingValidatorIndex,
+		CommitteeIndex:          3,
+		CommitteesAtSlot:        36,
+		CommitteeLength:         128,
+		ValidatorCommitteeIndex: 11,
```

Comment:
should not affect future forks

Reply:
If those fields are not used by our spec (and from my checks I believe there aren't) then I am not sure why we bother assigning them values.

I find adding code that looks somehow meaningful confusing.
I would delete this in another PR (because you copy-pasted existing logic, and we don't want to add changes here). 

In the meanwhile add the following comment.

// ISSUE 233: We are initializing unused struct fields here

Reply:
https://github.com/bloxapp/ssv-spec/pull/223/commits/2ca1614740174e179a7795618e0e2c074a8055b3